src/chain_sync/tipset_syncer.rs

```diff
@@ -695,9 +693,10 @@ where
 
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
         loop {
-            match self.as_mut().tipset_tasks.as_mut().poll_next(cx) {
-                Poll::Ready(Some(Ok(_))) => continue,
-                Poll::Ready(Some(Err(e))) => return Poll::Ready(Err(e)),
+            match self.as_mut().tipset_tasks.poll_join_next(cx) {
+                Poll::Ready(Some(Ok(Ok(_)))) => continue,
+                Poll::Ready(Some(Ok(Err(e)))) => return Poll::Ready(Err(e)),
+                Poll::Ready(Some(Err(e))) => panic!("{e}"),
```

Comment:
Propagate the panic. See: https://docs.rs/tokio/latest/tokio/task/struct.JoinError.html#method.into_panic

Reply:
If the tasks are canceled, I would log it and return normally.

Reply:
Fixed as disccussed